pkgs/development/python-modules/pyrsistent/default.nix

```diff
@@ -15,11 +16,13 @@ buildPythonPackage rec {
     sha256 = "5a31f6b093da3401fefdeb53a0980e3145bb9d2bf852b579cc7b39c7f0016c87";
   };
 
+  nativeBuildInputs = [ pytestrunner ];
```

Comment:
I may be missing something, but why isn't `pytestrunner` part of `checkInputs`?

Reply:
I think it's because it's necessary even if you're not running tests, just to run `setup.py`. It's in `setup_requires` [here](https://github.com/tobgu/pyrsistent/blob/b3d0c27554595a1c1edc781adefcf029632bdb45/setup.py#L82) and it fails the build when I move it to `checkInputs` if I also set `doCheck = false;`.

Reply:
I'm not sure I understand the issue. `pytestrunner` is necessary to build the derivation regardless of whether or not you're running tests, but not to run the executable, so isn't `nativeBuildInputs` exactly where it should go?

Are you saying we should write a patch that makes `pytestrunner` only necessary when you're running tests so that we can move it to `checkInputs`?

Reply:
Why do you need `pytestrunner` to build this package? If it's only listed in `setup.py` to ensure it's installed for the tests I'd propose to create a patch which removes that dependency. Then you can list `pytestrunner` in `checkInputs` (where it belongs IMHO) and nothing breaks if the tests need to be disabled in the future.

Or am I missing something?

Reply:
> Or am I missing something?

No @Ma27 I think you have it correct. I'll make this change.

pytestrunner even talks about achieving a [conditional requirement](https://github.com/pytest-dev/pytest-runner#conditional-requirement).

Reply:
https://github.com/NixOS/nixpkgs/pull/53599/commits/b8a6d1be56686ffb0a860b5184fd96b025150c05

Reply:
I realised now, that I do that in my Python packages, doing conditional requirement for pytestrunner in the `setup.py`. Although I wonder why is pytestrunner needed in `setup_requires` anyway? Why not just `test_requires`. Maybe some technical limitation in pytestrunner/setup.py.